render_silhouette_regmarks.py

```diff
@@ -0,0 +1,153 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA.
+#
+"""
+Base module for rendering regmarks for Silhouette CAMEO products in Inkscape.
+"""
+
+import sys, os
+# Enables stand alone mode and helps for tests #
+# We append the directory where this script lives and inkscape extension folder to sys.path
+sys.path.append(os.path.dirname(os.path.abspath(sys.argv[0])))
+sys_platform = sys.platform.lower()
+if sys_platform.startswith("win"):
+    sys.path.append(r"C:\Program Files\Inkscape\share\inkscape\extensions")
+elif sys_platform.startswith("darwin"):
+    sys.path.append("/Applications/Inkscape.app/Contents/Resources/share/inkscape/extensions")
+else:   # linux
+    sys.path.append("/usr/share/inkscape/extensions")
+
+import inkex
+from inkex import EffectExtension, Boolean, Rectangle, PathElement, Layer, Group, TextElement, Transform, BaseElement, SvgDocumentElement
+from gettext import gettext
+from inkex.units import convert_unit
+
+# Temporary Monkey Backport Patches to support functions that exist only after v1.2
+# TODO: If support for Inkscape v1.1 is dropped then this backport can be removed
+if not hasattr(inkex, "__version__") or inkex.__version__[0:3] < "1.2":
+	SvgDocumentElement.viewport_width = property(lambda self: convert_unit(self.get("width"), "px") or self.get_viewbox()[2])
+	SvgDocumentElement.viewport_height = property(lambda self: convert_unit(self.get("height"), "px") or self.get_viewbox()[3])
+	SvgDocumentElement._base_scale = lambda self, unit="px": (convert_unit(1, unit) or 1.0) if not all(self.get_viewbox()[2:]) else max([convert_unit(self.viewport_width, unit) / self.get_viewbox()[2], convert_unit(self.viewport_height, unit) / self.get_viewbox()[3]]) or convert_unit(1, unit) or 1.0
+	SvgDocumentElement.to_dimensional = staticmethod(lambda self, value, to_unit="px": convert_unit(value, to_unit))
+	SvgDocumentElement.to_dimensionless = staticmethod(lambda self, value: convert_unit(value, "px"))
+	SvgDocumentElement.equivalent_transform_scale = property(lambda self: max([self.to_dimensional(self.viewport_width, unit) / self.viewbox_width, self.to_dimensional(self.viewport_height, unit) / self.viewbox_height]) or 1.0)
+	SvgDocumentElement.viewport_to_unit = staticmethod(lambda self, value, unit="px": self.to_dimensional(self.to_dimensionless(value) / self.root.equivalent_transform_scale, unit))
```

Comment:
this error was because you've put self arg for static methods
and watch out `equivalent_transform_scale` relies on `scale` which calls `_base_scale()` which already has a sophisticated logic backported ...
```
if not hasattr(inkex, "__version__") or inkex.__version__[0:3] < "1.2":
    SvgDocumentElement.viewport_width = property(lambda self: convert_unit(self.get("width"), "px") or self.get_viewbox()[2])
    SvgDocumentElement.viewport_height = property(lambda self: convert_unit(self.get("height"), "px") or self.get_viewbox()[3])
    SvgDocumentElement._base_scale = lambda self, unit="px": (convert_unit(1, unit) or 1.0) if not all(self.get_viewbox()[2:]) else max([convert_unit(self.viewport_width, unit) / self.get_viewbox()[2], convert_unit(self.viewport_height, unit) / self.get_viewbox()[3]]) or convert_unit(1, unit) or 1.0
    BaseElement.to_dimensional = staticmethod(lambda value, to_unit="px": convert_unit(value, to_unit))
    BaseElement.to_dimensionless = staticmethod(lambda value: convert_unit(value, "px"))
    BaseElement.viewport_to_unit = lambda self, value, unit="px": self.to_dimensional(self.to_dimensionless(value) / self.root._base_scale(), unit)
```

... but finally it turns out it will fail on `PathElement.new(Path(...))` for some TypeErrors ... in parse_string() calling strargs() and probably more to come yet ...

NB: This is why I prefer **test driven development** and include tests right from the start :smile: 

so you may drop all those monkeypatches here and InsertRegmarks is only supported for inkex 1.2+ :+1: 
PS: maybe you can profit from some of those monkeypatches for #253